src/zenml/stack_stores/sql_stack_store.py

```diff
@@ -245,6 +245,66 @@ def register_stack_component(
             session.add(new_component)
             session.commit()
 
+    def update_stack_component(
+        self,
+        name: str,
+        component_type: StackComponentType,
+        component: StackComponentWrapper,
+    ) -> Dict[str, str]:
+        """Update a stack component.
+
+        Args:
+            name: The original name of the stack component.
+            component_type: The type of the stack component to update.
+            component: The new component to update with.
+
+        Raises:
+            KeyError: If no stack component exists with the given name.
+        """
+        with Session(self.engine) as session:
+            updated_component = session.exec(
+                select(ZenStackComponent)
+                .where(ZenStackComponent.component_type == component_type)
+                .where(ZenStackComponent.name == name)
+            ).first()
+
+            if not updated_component:
+                raise KeyError(
+                    f"Unable to update stack component (type: {component.type}) "
+                    f"with name '{component.name}': No existing stack component "
+                    f"found with this name."
+                )
+
+            updated_component.configuration = component.config
+
+            # handle any potential renamed component
+            updated_component.name = component.name
+
+            # rename components inside stacks
+            updated_stack_definitions = session.exec(
+                select(ZenStackDefinition)
+                .where(ZenStackDefinition.component_type == component_type)
+                .where(ZenStackDefinition.component_name == name)
+            ).all()
+            for stack_definition in updated_stack_definitions:
+                stack_definition.component_name = component.name
+                session.add(stack_definition)
+
+            session.add(updated_component)
+            session.commit()
+        logger.info(
+            "Updated stack component with type '%s' and name '%s'.",
+            component_type,
+            component.name,
+        )
+        # tc = session.exec(
```

Comment:
Can probably be removed? Similar to the local store, I think we should handle a component with that name already existing